### PR TITLE
Adds the concept of recipients to Messaages, MessageServlets, and DataStore

### PR DESCRIPTION
**Pull request by prudencep**
Added a recipient variable to Message.java and changed the position of the curly brackets because I like seeing the blocks of code properly.

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Message.java [hunk 1]
```diff
@@ -19,41 +19,56 @@
...
 
-  public long getTimestamp() {
+  public long getTimestamp()
+  {
     return timestamp;
   }
+
+  public String getRecipient()
+  {
+    return recipient; //added accessor to recipient
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/data/Message.java]: This comment is unnecessary. Refrain from comments that are self explanatory in code. Use comments in code for complex logic that someone reading the code might not understand at a glance. Here it is clear the getRecipient() returns a recipient and that the method name "get" refers to an accessor.

File: src/main/java/com/google/codeu/data/Message.java [hunk 2]
```diff
@@ -19,41 +19,56 @@
...
   }
 
-  public Message(UUID id, String user, String text, long timestamp) {
+  public Message(UUID id, String user, String text, long timestamp)
+  {
     this.id = id;
     this.user = user;
     this.text = text;
     this.timestamp = timestamp;
+    this.recipient = recipient; //set recipient in default constructor
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/data/Message.java]: This comment is unnecessary. It is clear that we are setting the recipient in the constructor. The other variables do not have a similar comment.

File: src/main/java/com/google/codeu/data/Message.java [hunk 3]
```diff
@@ -19,41 +19,56 @@
...
   private long timestamp;
+  private String recipient; //added recipient for messages
 
   /**
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
    * random ID and uses the current system time for the creation time.
    */
-  public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis());
+  public Message(String user, String text, String recipient)/*Added recipient to list of constructors*/
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/data/Message.java]: Refrain from commenting in line. Comments are usually reserved their own line. Also refrain from writing comments that someone reading your code can clearly see what the code is doing. Notice that parts of original code that do and do not have comments. It is important to follow what the original code does and make sure your code is integrated with the style this file comes with.

File: src/main/java/com/google/codeu/data/Message.java [hunk 4]
```diff
@@ -19,41 +19,56 @@
...
 /** A single message posted by a user. */
-public class Message {
+public class Message
+{
 
   private UUID id;
   private String user;
   private String text;
   private long timestamp;
+  private String recipient; //added recipient for messages
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/data/Message.java]: ditto on the inline comment (read messages below)

**Comment by FluffySheep** [comment]
Change title of pull request to: Added a recipient variable to the Message class.

It is unclear just from browsing the pull requests where the recipient variable is added to. The description should indicate why you added the recipient variable (which might include a link to the instructions).

Also, you'll have to put the curly braces back on the same line, sorry :(. It is part of the google java style guide which our stickler-ci checker will continue to flag and won't let you merge in code. This is the same way as in Google as well. We have something like a checker that will prevent you from submitting if you don't follow the style guide.

File: src/main/java/com/google/codeu/data/Message.java (hunk 1 above)
Comment (prudencep) [reply src/main/java/com/google/codeu/data/Message.java]: Removed. I'll take note of this

File: src/main/java/com/google/codeu/data/Message.java (hunk 2 above)
Comment (prudencep) [reply src/main/java/com/google/codeu/data/Message.java]: Removed this too.

File: src/main/java/com/google/codeu/data/Message.java (hunk 3 above)
Comment (prudencep) [reply src/main/java/com/google/codeu/data/Message.java]: Ok noted!

**Comment by prudencep** [comment]
I have now fixed all the errors on this branch.

**Comment by FluffySheep** [comment]
You are now welcome to merge this pull request into develop. Great work!

**Pull request merged**